--- targetlibs/nrf5x_15/components/drivers_nrf/usbd/nrf_drv_usbd_errata.h	2018-03-22 15:25:08.000000000 +0000
+++ targetlibs/nrf5x_15/components/drivers_nrf/usbd/nrf_drv_usbd_errata.h	2020-06-10 10:36:05.840257060 +0100
@@ -93,8 +93,10 @@
  */
 static inline bool nrf_drv_usbd_errata_type_52840_fp1(void)
 {
+
+  // https://devzone.nordicsemi.com/f/nordic-q-a/34605/usb-not-enumerating-after-reset/133051#133051
     return ( nrf_drv_usbd_errata_type_52840() &&
-               ( ((*(uint32_t *)0xF0000FE8) & 0xF0) == 0x10 ) &&
+               ( ((*(uint32_t *)0xF0000FE8) & 0xF0) == 0x20 ) && // note: modified from 0x10 to 0x20!
                ( ((*(uint32_t *)0xF0000FEC) & 0xF0) == 0x00 ) );
 }
 
