--- a/targetlibs/nrf5x_15/components/ble/ble_services/ble_nus/ble_nus.c	2021-01-20 09:24:58.258245089 +0000
+++ b/targetlibs/nrf5x_15/components/ble/ble_services/ble_nus/ble_nus.c	2020-05-13 10:28:38.926572968 +0100
@@ -233,8 +233,13 @@
 
     memset(&cccd_md, 0, sizeof(cccd_md));
 
+    if (p_nus_init->encrypt) {
+      BLE_GAP_CONN_SEC_MODE_SET_ENC_NO_MITM(&cccd_md.read_perm);
+      BLE_GAP_CONN_SEC_MODE_SET_ENC_NO_MITM(&cccd_md.write_perm);
+    } else {
     BLE_GAP_CONN_SEC_MODE_SET_OPEN(&cccd_md.read_perm);
     BLE_GAP_CONN_SEC_MODE_SET_OPEN(&cccd_md.write_perm);
+    }
 
     cccd_md.vloc = BLE_GATTS_VLOC_STACK;
 
