diff -uN -r tensorflow/lite/experimental/micro/tools/make/gen/make_x86_64/prj/hello_world/make/tensorflow/lite/micro/kernels/all_ops_resolver.h /home/gw/workspace/Espruino/libs/tensorflow/tensorflow/lite/micro/kernels/all_ops_resolver.h
--- a/libs/tensorflow/tensorflow/lite/micro/kernels/all_ops_resolver.h	2019-09-11 15:49:01.989141904 +0100
+++ b/libs/tensorflow/tensorflow/lite/micro/kernels/all_ops_resolver.h	2019-09-11 15:46:24.455056488 +0100
@@ -9,6 +9,7 @@
 See the License for the specific language governing permissions and
 limitations under the License.
 ==============================================================================*/
+// Minor modifications made for Espruino Microcontroller build by Gordon Williams <gw@pur3.co.uk>
 #ifndef TENSORFLOW_LITE_MICRO_KERNELS_ALL_OPS_RESOLVER_H_
 #define TENSORFLOW_LITE_MICRO_KERNELS_ALL_OPS_RESOLVER_H_
 
@@ -22,9 +23,8 @@
 class AllOpsResolver : public MicroMutableOpResolver {
  public:
   AllOpsResolver();
-
- private:
   TF_LITE_REMOVE_VIRTUAL_DELETE
+
 };
 
 }  // namespace micro

diff -uN -r tensorflow/lite/experimental/micro/tools/make/gen/make_x86_64/prj/hello_world/make/tensorflow/lite/micro/micro_error_reporter.h /home/gw/workspace/Espruino/libs/tensorflow/tensorflow/lite/micro/micro_error_reporter.h
--- a/libs/tensorflow/tensorflow/lite/micro/micro_error_reporter.h	2019-09-11 15:49:01.969141639 +0100
+++ b/libs/tensorflow/tensorflow/lite/micro/micro_error_reporter.h	2019-09-11 15:46:25.167065911 +0100
@@ -12,6 +12,7 @@
 See the License for the specific language governing permissions and
 limitations under the License.
 ==============================================================================*/
+// Minor modifications made for Espruino Microcontroller build by Gordon Williams <gw@pur3.co.uk>
 #ifndef TENSORFLOW_LITE_MICRO_MICRO_ERROR_REPORTER_H_
 #define TENSORFLOW_LITE_MICRO_MICRO_ERROR_REPORTER_H_
 
@@ -26,8 +27,6 @@
  public:
   ~MicroErrorReporter() {}
   int Report(const char* format, va_list args) override;
-
- private:
   TF_LITE_REMOVE_VIRTUAL_DELETE
 };

diff --git a/libs/tensorflow/tensorflow/lite/micro/micro_mutable_op_resolver.h b/libs/tensorflow/tensorflow/lite/micro/micro_mutable_op_resolver.h
index 21066cf4..3428d5f5 100644
--- a/libs/tensorflow/tensorflow/lite/micro/micro_mutable_op_resolver.h
+++ b/libs/tensorflow/tensorflow/lite/micro/micro_mutable_op_resolver.h
@@ -12,6 +12,7 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 ==============================================================================*/
+// Minor modifications made for Espruino Microcontroller build by Gordon Williams <gw@pur3.co.uk>
 #ifndef TENSORFLOW_LITE_MICRO_MICRO_MUTABLE_OP_RESOLVER_H_
 #define TENSORFLOW_LITE_MICRO_MICRO_MUTABLE_OP_RESOLVER_H_
 
@@ -96,7 +97,7 @@ class MicroOpResolver : public OpResolver {
  private:
   TfLiteRegistration registrations_[tOpCount];
   unsigned int registrations_len_ = 0;
-
+ public:
   TF_LITE_REMOVE_VIRTUAL_DELETE
 };
 

