--- a/tensorflow/lite/experimental/micro/memory_planner/greedy_memory_planner.h	2019-09-11 09:14:45.509106591 +0100
+++ b/tensorflow/lite/experimental/micro/memory_planner/greedy_memory_planner.h	2019-09-11 09:13:54.204422772 +0100
@@ -17,6 +17,7 @@
 #define TENSORFLOW_LITE_EXPERIMENTAL_MICRO_MEMORY_PLANNER_GREEDY_MEMORY_PLANNER_H_
 
 #include "tensorflow/lite/experimental/micro/memory_planner/memory_planner.h"
+#include "tensorflow/lite/experimental/micro/compatibility.h"
 
 namespace tflite {
 
@@ -125,6 +126,8 @@
 
   // Whether buffers have been added since the last plan was calculated.
   bool need_to_calculate_offsets_;
+
+  TF_LITE_REMOVE_VIRTUAL_DELETE
 };
 
 }  // namespace tflite
